src/main/java/com/bravo/user/validator/AddressValidator.java

```diff
@@ -0,0 +1,34 @@
+package com.bravo.user.validator;
+
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+
+import com.bravo.user.exception.BadRequestException;
+import com.bravo.user.model.dto.AddressReadDto;
+import com.bravo.user.utility.ValidatorUtil;
+
+@Component
+public class AddressValidator extends CrudValidator {
+
+  public void validateId(String id){
+    if(ValidatorUtil.isInvalid(id)){
+      throw new BadRequestException("'id' is required");
+    }
+  }
+
+  @Override
+  protected void validateRetrieve(Object o, Errors errors) {
```

Comment:
This method is currently not being used, yet if it were to be used by the Retrieve implementation there are a couple issues with it. 

1. The Object o would actually be an instance of the `AddressFilter` not the `AddressReadDto`.
2. There should be no validation on required fields for a retrieve function since the default functionality in the `AbstractSpecification` is to return all the data (paginated) when there are no matches.